rfcs/0001-tokens-naming-convention.md

```diff
@@ -0,0 +1,406 @@
+```
+color.action.background
+color.feedback.text
+font.heading.size
+font.body.weight
+```
+
+#### TL;DR
+**Base** refers to a token’s backbone that combines
+  1. **category** - `color`, `space`, `size`, `font` etc.
+  2. **behavior** - `action`, `feedback` etc.
+  3. **property** - `size`, `weight`, `border` etc.
+> 📝 Note: It's not mandatory to use all the three sub levels of the `Base` i.e `category.behavior.property` always. You can use them as per the use case by skipping some of the grouping levels if required.
+### Modifier
+Whenever we will build components we will always come across use cases like defining token for a particular `variant`(primary, secondory, success, error, information etc.) of a component in a particular `state`(hover, click, active, state etc.). 
+
+To achieve the above use case we will use modifiers which can further be categorised into `variant`, `state`, `scale`, and `mode`. 
+
+Modifiers will be the last level in our token naming hierarchy.
+
+Modifiers can be used independently or can be paired with levels like category, behavior and property to form a purposeful decision(a.k.a meaningful self explanatory token). For eg: 
+- `Button.color.text.primary`
+  - `Button` - Object
+  - `color` - Base(category)
+  - `text` - Base(property)
+  - `primary` - Modifier(variant)
+- `theme.color.action.background.hover`
+  - `theme` - Object
+  - `color` - Base(category)
+  - `action` - Base(behavior)
+  - `background` - Base(property)
+  - `hover` - Modifier(state)
+
+#### Variant
+A variant distinguishes alternative use cases.
+
+For example different variants of a `Button`
+* `primary`
+* `secondary`
+* `tertiary`
+
+Similarly, different variant of `Notification`
+* `success` - to indicate a successful action
+* `error` - to indicate that an action errored out
+* `information` - to nudge the user with information
+* `warning` - to indicate a warning
+
+For example: `theme.color.action.text.primary`
+#### State
+Tokens can specify properties based on interactive states, like:
+* `default`
+* `hover` - when a pointer is positioned above an object
+* `click` - when a user presses an object
+* `focus` - when an object is able to accept input
+* `disabled` - when an object is not able to accept input
+* `visited` - when a link is already visited
+* `error` - when an object is in an error state
+
+For example: `theme.color.action.text.primary.focus`
+
+#### Scale
+Scale serves as a mofifer to create more granular token name in scenarios where we want to create a token which work differently on different scales
+
+* Linear Scale: `1`, `2`, `3`, `4` and so on
+* T-shirt Size Scale: `s`, `m`, `l`, `xl` and so on
+
+For example: `theme.font.size.s`, `Button.space.paddingLeft.primary.hover.m`
+
+#### Mode
+Mode is usually a modifier to help us create tokens to distinguish values across two or more surface/background settings on which elements appear 
```

Comment:
Where can we use this? Like if we are already using `theme` as an `object`? Or this is something inside of `theme`?

Reply:
inside theme. Check `Principles` section